tests/mir-opt/reference_prop.rs

```diff
@@ -0,0 +1,103 @@
+// unit-test PropagateReferences
+
+#[inline(never)]
+fn opaque(_: impl Sized) {}
+
+macro_rules! case {
+    ($fn_name:ident: $($mut:tt)?) => {
+        fn $fn_name<'a, T: Copy>(single: &'a $($mut)? T, mut multiple: &'a $($mut)? T) {
```

Comment:
Can this be split up into different functions? It's quite hard to tell what corresponds to what